assembl/static2/js/app/components/cookies/acceptCookiesModal.jsx

```diff
@@ -0,0 +1,184 @@
+    const lastRouteString = getRouteLastString(pathname);
+    const isOnLegalContentPage = legalContentSlugs.includes(lastRouteString);
+    let userHasAcceptedAllLegalContents;
+    // This array gathers all the legal contents to accept by their 'ACCEPT_...' formatted name
+    const legalContentsToAcceptByCookieName = this.getLegalContentsToAccept();
+    if (userId) {
+      userHasAcceptedAllLegalContents = legalContentsToAcceptByCookieName.every(legalContent =>
+        acceptedLegalContentsList.includes(legalContent)
+      );
+    }
+    // The modal is only showed to a user who is connected but hasn't yet accepted legal contents and isn't currently reading them
+    if (!isOnLegalContentPage && !userHasAcceptedAllLegalContents && userId) {
+      this.setState({ showModal: true });
+    }
+  };
+
+  handleModalCheckbox = () => {
+    this.setState(prevState => ({
+      modalIsChecked: !prevState.modalIsChecked
+    }));
+  };
+
+  getLegalContentsToAccept = (): LegalContentsArray => {
+    const { hasTermsAndConditions, hasPrivacyPolicy, hasUserGuidelines } = this.props;
+    const legalContentsToAccept = {
+      ACCEPT_CGU: hasTermsAndConditions,
+      ACCEPT_PRIVACY_POLICY_ON_DISCUSSION: hasPrivacyPolicy,
+      ACCEPT_USER_GUIDELINE_ON_DISCUSSION: hasUserGuidelines
+    };
+    const filteredLegalContentsToAccept = LEGAL_CONTENTS_TO_ACCEPT.filter(contentType => legalContentsToAccept[contentType]);
+    return filteredLegalContentsToAccept;
+  };
+
+  acceptAllLegalContents = () => {
+    const legalContentsToAccept = this.getLegalContentsToAccept();
+    this.props.updateAcceptedCookies({
+      variables: { actions: legalContentsToAccept }
+    });
+  };
+
+  closeModal = () => {
+    this.setState({ showModal: false });
+  };
+
+  render() {
+    const { hasTermsAndConditions, hasPrivacyPolicy, hasUserGuidelines } = this.props;
+    const { showModal, modalIsChecked } = this.state;
+    const slug = getDiscussionSlug();
+    const legalContentsToAcceptByRouteName = {
+      terms: hasTermsAndConditions,
+      privacyPolicy: hasPrivacyPolicy,
+      userGuidelines: hasUserGuidelines
+    };
+    const legalContentsArray = Object.keys(legalContentsToAcceptByRouteName).map(
+      key => (legalContentsToAcceptByRouteName[key] ? key : null)
+    );
+    // This array gathers all the legal contents to accept by their route name
+    const cleanLegalContentsArray = legalContentsArray.filter(el => el !== null);
+
+    return cleanLegalContentsArray ? (
+      <Modal show={showModal} backdrop="static">
+        <Modal.Header>
+          <Modal.Title>
+            <Translate value="legalContentsModal.title" />
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <FormGroup className="justify">
+            <Checkbox onChange={this.handleModalCheckbox} checked={modalIsChecked} type="checkbox" inline>
+              <Translate value="legalContentsModal.iAccept" /> {/* $FlowFixMe */}
```

Comment:
What is the flow error?

Reply:
It thinks cleanLegalContentsArray can be null but the whole return is not happening if it is...